waltid-services/waltid-verifier-api/src/main/kotlin/id/walt/verifier/oidc/OIDCVerifierService.kt

```diff
@@ -206,7 +206,8 @@ object OIDCVerifierService : OpenIDCredentialVerifier(
         val parsedDeviceResponse = DeviceResponse.fromCBORBase64URL(tokenResponse.vpToken!!.jsonPrimitive.content)
         val parsedMdoc = parsedDeviceResponse.documents[0]
         val deviceKey = OneKey(CBORObject.DecodeFromBytes(parsedMdoc.MSO!!.deviceKeyInfo.deviceKey.toCBOR()))
-        val issuerKey = parsedMdoc.issuerSigned.issuerAuth?.x5Chain?.let { X509CertUtils.parse(it) }?.publicKey ?: throw Exception("Issuer key not found in x5Chain header (33)")
+        val issuerKey = parsedMdoc.issuerSigned.issuerAuth?.x5Chain?.let { X509CertUtils.parse(it) }?.publicKey
+            ?: throw BadRequestException("Issuer key not found in x5Chain header (33)")
```

Comment:
No idea , seems like something too technical

Reply:
Absolutely. 

@cpatsonakis , could you help us? What is this error about? Is it a problem caused by the user or an internal issue? I mean, is it something the user has control over? When he gets this error message, he knows what to do?

Reply:
I am not knowledgeable on mDocs, however, what I can say is that it attempts to parse the credential issuer's key from the `x5c` JWS header to verify the signature of the mDoc credential.

Reply:
maybe a message like this :` Issuer's Public Key Missing: The x5c header in the JWT is either missing or does not contain the expected X.509 certificate chain. Please ensure that the x5c header is correctly formatted and includes the issuer’s public key.` is more clear ?

Reply:
It's much more clear. But, when you say "ensure that the x3c header is correctly formatted and includes the issuer's public key", is it something that needs to be done on the client (i.e. user's call) or on the server (Issuer API) side?

Reply:
im not sure , @severinstampler can you please help us with this ?

Reply:
it is configured in the body request of the issuer api as  `x5Chain`

Reply:
The mdoc is signed by the issuer using the COSE spec (https://datatracker.ietf.org/doc/html/rfc8152), and the public key of the issuer is included in the mdoc with the x509 certificate chain, which the verifier can use to first verify the validity of the public key and then verify the signature.
Now in this case, if the x509 chain is invalid, there is nothing the verifier can do, as the mdoc is invalid. And will need to be re-issued by the issuer.
It may, however, be of interest to the issuer to get information about what or why the mdoc is invalid, so that a potential bug can be fixed on the issuer side.